From 98e1eb777717e331f2c7b831c14caef853ce9169 Mon Sep 17 00:00:00 2001
From: Jeevaka Prabu Badrappan <jeevaka.badrappan@intel.com>
Date: Thu, 10 Nov 2022 15:49:20 +0530
Subject: [PATCH] Revert "[Decode] Enhance robustness of ddi for AV1"

In few laptop clients, camera preview is stuck after first
few frames.

As the commit d4b51b5e133c60742e3ec43012bac6aa88610969 is for
AV1 and AV1 not supported in Android 11 commit reverted to
fix the camera preview stuck issue.

Change-Id: I683da1d036783dabca1b18c2bade9f882b56a8a0
Tracked-On: OAM-104651
Signed-off-by: Jeevaka Prabu Badrappan <jeevaka.badrappan@intel.com>
---
 media_driver/linux/common/codec/ddi/media_libva_decoder.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/media_driver/linux/common/codec/ddi/media_libva_decoder.cpp b/media_driver/linux/common/codec/ddi/media_libva_decoder.cpp
index c15c74d15..b5fc343be 100755
--- a/media_driver/linux/common/codec/ddi/media_libva_decoder.cpp
+++ b/media_driver/linux/common/codec/ddi/media_libva_decoder.cpp
@@ -1233,7 +1233,7 @@ VAStatus DdiDecode_StatusReport(PDDI_MEDIA_CONTEXT mediaCtx, DecodePipelineAdapt
                 PDDI_MEDIA_SURFACE_HEAP_ELEMENT mediaSurfaceHeapElmt = (PDDI_MEDIA_SURFACE_HEAP_ELEMENT)mediaCtx->pSurfaceHeap->pHeapBase;
 
                 uint32_t j = 0;
-                for (j = 0; j < mediaCtx->uiNumSurfaces; j++, mediaSurfaceHeapElmt++)
+                for (j = 0; j < mediaCtx->pSurfaceHeap->uiAllocatedHeapElements; j++, mediaSurfaceHeapElmt++)
                 {
                     if (mediaSurfaceHeapElmt != nullptr &&
                             mediaSurfaceHeapElmt->pSurface != nullptr &&
-- 
2.39.0

